adr/adr-responses-api.md

```diff
@@ -0,0 +1,52 @@
+* **Ecosystem Compatibility:** Our agentic capability will be natively compatible with the vast ecosystem of tools, libraries, and monitoring solutions that are built to interface with the OpenAI API format. This accelerates both internal development and customer adoption.
+* **Enhanced Flexibility and Control:** A client-side agent construct gives the application developer full control over the agent's execution loop, state, and tool implementation. This is critical for complex, mission-critical applications where logic cannot be a black box on a server.
+* **Clear Separation of Concerns:** The Llama Stack server is responsible for what it does best: executing LLM inference with tool-calling capabilities. The client is responsible for the application-level concern of orchestrating the agentic logic. This is a clean and scalable architectural pattern.
+* **Stateless Backend:** A stateless API is inherently more scalable, resilient, and easier to manage from an operations perspective. It eliminates concerns about server-side session memory, replication, and expiry.
+
+### Negative
+
+* **Implementation Overhead:** We assume the responsibility for implementing or integrating the client-side Agent construct. This requires dedicated engineering resources, whether for building, testing, or vetting and supporting a third-party solution.
+* **Increased Client-side Complexity:** The logic for managing the agent's state and execution loop now resides on the client. While our chosen Agent construct will abstract this away, the fundamental complexity is shifted from the Llama Stack server to the client. We must ensure our chosen solution is robust, performant, and easy to use.
+* **Difficulty in Sharing and Collaboration:** A server-side agent can be exposed as a persistent, shareable endpoint. A client-side agent, however, exists only within the context of the application that instantiated it. To "share" an agent, one must serialize its configuration and state, transfer it, and have the recipient re-instantiate a new agent from that data. This "snapshot-and-restore" process creates friction for use cases requiring pre-configured, easily shareable bots or live collaborative agents. (Note: The containerized API proposal could mitigate this, which will be analyzed in the next ADR).
+* **Potential for State Inconsistency:** As the client is the source of truth for the agent's state, developers using our solution must correctly manage this state. The design of our chosen Agent construct should provide strong guardrails to prevent common errors.
+* **Deferred Decision:** The critical decision on the form and implementation of the Agent construct must be resolved in a timely manner to avoid development bottlenecks.
```

Comment:
Do we talk about auth somewhere?

Reply:
No.  Is there anything in particular you want to address relating to auth?